### DISCUSSION TITLE
Linear solutions to a nonlinear equation

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#18417] [Q&A Modules: General] [answered]
I am applying the following two kernels, TimeDerivative and BodyForce to solve the equation d/dt(u)=-Cu. The problem is an initial condition problem. The numerical solution from MOOSE results in a linear decrease in u over the time period. The analytical solution results in a exponential decrease in u over time. Is there a better combination of kernels that should be applied in order to get the expected exponential decrease in u?

**Comment by GiudGiud** [comment]
Hello

I think you mean to use CoupledForce. BodyForce is just C not Cu

Guillaume

**Reply by TLWise** [reply]
Thank you for the assistance. I changed the kernel to a CoupledForce, the result was the same linear trend. I then added Diffusion of the coupled v value of the CoupledForce. The results was also the same as the previous two methods used.

**Reply by GiudGiud** [reply]
How many time steps are you taking?
Did you specify ‘v=u’ for the couples force kernel?

**Reply by TLWise** [reply]
Current 130 time steps. Have used up to 13000 time steps with the same result. I did not set v = u.

**Reply by TLWise** [reply]
Tried to define v = u. Got the following message: Coupled variable 'v' needs to be different from 'variable' with CoupledForce, consider using Reaction or somethig similar.

**Reply by hugary1995** [reply]
You should use the `CoefReaction` kernel for the reaction term. [https://mooseframework.inl.gov/moose/source/kernels/CoefReaction.html](https://mooseframework.inl.gov/moose/source/kernels/CoefReaction.html)

**Reply by TLWise** [reply]
That was the final missing piece, CoefReaction. Thank you both for helping me with this. The trend of the numerical solution is now in proper agreement with the analytical solution.

**Comment by TLWise** [comment, marked as answer]
Implementation of this provided the solutions to the nonlinear trend of the numerical solution:

 hugary1995:   You should use the CoefReaction kernel for the reaction term. https://mooseframework.inl.gov/moose/source/kernels/CoefReaction.html.

**Reply by GiudGiud** [reply]
I ll review the documentation to make sure everything is clear. There should not be a confusion between BodyForce/CoupledForce and Reaction.
